airflow/providers/databricks/operators/databricks.py

```diff
@@ -384,6 +386,11 @@ def _get_hook(self, caller: str) -> DatabricksHook:
         )
 
     def execute(self, context: Context):
+        if "pipeline_task" in self.json and self.json["pipeline_task"].get("pipeline_id") is None:
```

Comment:
You need to add an extra check on `pipeline_name`:
```suggestion
        if (
            "pipeline_task" in self.json 
            and self.json["pipeline_task"].get("pipeline_id") is None
            and self.json["pipeline_task"].get("pipeline_name")
        ):
```